src/sqlfluff/core/templaters/base.py

```diff
@@ -302,7 +302,10 @@ def templated_slice_to_source_slice(
                 return slice(insertion_point, insertion_point)
             # It's within a segment.
             else:
-                if ts_start_subsliced_file[0][0] == "literal":
+                if (
+                    ts_start_subsliced_file
```

Comment:
During work on this PR, at some point it was crashing here with an `IndexError` because `ts_start_subsliced_file` was empty. Without really understanding what was going on, I tried the simplest possible fix -- checking first that the array is not empty. Seemed to work.